design-documents/graph-ql/coverage/b2b/purchase-order.graphqls

```diff
@@ -0,0 +1,191 @@
+    comments(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderComments @doc(description: "Purchase order comments")
+    payment_methods: [PaymentMethod] @doc(description: "Payment details for the purchase order")
+    shipping_address: CustomerAddress @doc(description: "The shipping address for the purchase order")
+    billing_address: CustomerAddress @doc(description: "The billing address for the purchase order")
+    carrier: String @doc(description: "The shipping carrier for the purchase order delivery")
+    shipping_method: String @doc(description: "The delivery method for the purchase order")
+    items(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderItems @doc(description: "Items that belong to the purchase order")
+    history_log(currentPage: Int = 1, pageSize: Int = 20): PurchaseOrderHistoryLog @doc(description: "The log of the events related to the purchase order")
+    approval_flow: PurchaseOrderApprovalFlow @doc(description: "The log of the events related to the purchase order approval flow")
+    available_actions: [PurchaseOrderAction] @doc(description: "Purcahse order actions available to the customer. Can be used to display action buttons on the client")
+}
+
+enum PurchaseOrderAction {
+    REJECT
+    CANCEL
+    APPROVE
+}
+
+type PurchaseOrderApprovalFlow {
+    items: [PurchaseOrderApprovalFlowItem]!
+}
+
+type PurchaseOrderApprovalFlowItem {
+    uid: ID! @doc(description: "Unique identifier of the purchase order flow item.")
+    title: String! @doc(description: "Summary of the event related to purchase order approval flow")
+    description: String! @doc(description: "Description of the event related to purchase order approval flow")
+    status: PurchaseOrderApprovalFlowItemStatus! @doc(description: "Status associated with the event related to purchase order approval flow")
+}
+
+enum PurchaseOrderApprovalFlowItemStatus {
+    PENDING
+    APPROVED
+    REJECTED
+}
+
+type PurchaseOrderComments {
+    items: [PurchaseOrderComment]!
+    page_info: SearchResultPageInfo
+    total_count: Int
+}
+
+type PurchaseOrderComment {
+    uid: ID! @doc(description: "Unique identifier of the comment.")
+    timestamp: String! @doc(description: "The date and time when the comment was created")
+    author: String! @doc(description: "The name of the user who left the comment")
+    text: String! @doc(description: "The text of the comment")
+}
+
+type PurchaseOrderHistoryLog {
+    items: [PurchaseOrderHistoryItem]!
+    page_info: SearchResultPageInfo
+    total_count: Int
+}
+
+type PurchaseOrderHistoryItem {
+    uid: ID! @doc(description: "Unique identifier of the purchase rder history item.")
+    timestamp: String! @doc(description: "The date and time when the event happened.")
+    description: String! @doc(description: "Description of the event.")
+}
+
+type PurchaseOrderItems {
+    items: [PurchaseOrderItemInterface]!
+    page_info: SearchResultPageInfo
+    total_count: Int
```

Comment:
If `items` is non-nullable, `total_count` should be too. Would be weird to have `items.length == 0` and `total_count == null`